modules/category/category.service.js

```diff
@@ -26,61 +26,73 @@ class CategoryService {
     throw new Error(CATEGORY_NOT_FOUND);
   }
 
-  async updateCategory(id, category, upload) {
-    await this.getCategoryById(id);
+  async updateCategory({ id, category, upload }) {
+    const categoryToUpdate = await Category.findById(id);
+    if (!categoryToUpdate) {
+      throw new Error(CATEGORY_NOT_FOUND);
+    }
+
     if (await this.checkCategoryExist(category, id)) {
       throw new Error(CATEGORY_ALREADY_EXIST);
     }
-    if (upload) {
-      await deleteFiles(
-        Object.values(category.images).filter(
-          item => typeof item === 'string' && item
-        )
-      );
-      const uploadResult = await uploadFiles([upload]);
-      const imageResults = await uploadResult[0];
-      category.images = imageResults.fileNames;
+
+    if (!upload || !Object.keys(upload).length) {
+      return await Category.findByIdAndUpdate(id, category, { new: true });
     }
-    return await Category.findByIdAndUpdate(id, category, {
-      new: true,
-    });
+    const uploadResult = await uploadFiles([upload]);
+
+    const uploadResults = await uploadResult[0];
```

Comment:
If we pass more than one image it won't add it to category result.

Reply:
why do we need  pass  more than one image?